lib/generator/template_values.rb

```diff
@@ -1,24 +1,30 @@
 module Generator
   # Contains methods accessible to the ERB template
   class TemplateValues
-    attr_reader :abbreviated_commit_hash, :version, :test_cases
+    attr_reader :abbreviated_commit_hash, :version, :exercise_name, :test_cases
```

Comment:
It took me a while to work out why this was needed, a commit message comment could be good here, although it probably doesn't mater since I'm probably the only one who would read it/care.

The template will need the exercise_name so it can do the correct class name substitutions.

Here is a wacky thought, do the ClassNames need to be ProblemSpecific? 

Currently I think the answer is yes, but philosophically, I suspect not.